testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java

```diff
@@ -93,11 +98,51 @@ protected abstract QueryRunner createQueryRunner()
     public final void close()
             throws Exception
     {
-        afterClassCloser.close();
-        queryRunner = null;
-        h2QueryRunner = null;
-        sqlParser = null;
-        queryAssertions = null;
+        try {
+            try {
+                checkQueryMemoryReleased();
+            }
+            catch (Throwable e) {
+                closeAllSuppress(e, afterClassCloser);
+                throw e;
+            }
+
+            afterClassCloser.close();
```

Comment:
Should this go to `finally`?

Reply:
no. It would mess up exception in case closer would throw. I want to get exception from `checkQueryMemoryReleased` if there is one.

Reply:
It's still feels like it is worth calling `close`. If `close` is not called the in memory cluster will remain active while the testing framework is running other test cases. Maybe put it in `finally` and wrap it in a `try-catch`?

Reply:
i agree. `afterClassCloser` must be invoked even if `checkQueryMemoryReleased` fals.

Reply:
But it is. In `catch` clause

Reply:
right, missed that.
you could also use `try()` for same effect

Reply:
true. Wil change

Reply:
You meant like that?

Reply:
I totally missed that the close was called. The current state looks good to me.